src/main/java/build/buildfarm/cas/cfc/CASFileCache.java

```diff
@@ -1458,7 +1442,7 @@ private void processRootFile(
           });
     }
 
-    joinThreads(pool, "Populating Directories...", 1, MINUTES);
```

Comment:
why hard-code this further down the callstack?

Reply:
Just to remove duplication and put common code (time and unit in this case) into the common method.